### PR TITLE
[network-data] fix compile error when building service without border router

### PR DESCRIPTION
**Pull request by dismirlian**

### CODE REVIEW

File: src/core/thread/network_data_local.cpp [hunk 1]
```diff
@@ -370,8 +370,11 @@ otError Local::SendServerDataNotification(void)
 #if OPENTHREAD_CONFIG_BORDER_ROUTER_ENABLE
         !IsOnMeshPrefixConsistent() || !IsExternalRouteConsistent()
 #endif
+#if OPENTHREAD_CONFIG_BORDER_ROUTER_ENABLE && OPENTHREAD_CONFIG_TMF_NETDATA_SERVICE_ENABLE
```
Comment (jwhui) [inline src/core/thread/network_data_local.cpp]: Thanks for identifying this issue and proposing a fix! Would it be cleaner to simply break this `VerifyOrExit()` up into two separate statements to simplify the `#if`s?

Comment (dismirlian) [reply src/core/thread/network_data_local.cpp]: Hi @jwhui, I agree, much cleaner.

**Comment by dismirlian** [comment]
I'll close and re-open to kick the failed Travis CI again.

File: src/core/thread/network_data_local.cpp (hunk 1 above)
Comment (jwhui) [reply src/core/thread/network_data_local.cpp]: @dismirlian, I think you want something like:
```c++
    bool isConsistent =	true;

#if OPENTHREAD_CONFIG_BORDER_ROUTER_ENABLE
    isConsistent &= IsOnMeshPrefixConsistent() && IsExternalRouteConsistent();
#endif
#if OPENTHREAD_CONFIG_TMF_NETDATA_SERVICE_ENABLE
    isConsistent &= IsServiceConsistent();
#endif

    VerifyOrExit(!isConsistent, ClearResubmitDelayTimer());
```
This should help resolve the Travis test failures.

Comment (dismirlian) [reply src/core/thread/network_data_local.cpp]: @jwhui, I'm sorry, I thought it was the usual Travis timeouts... I'll check the Travis failures and your suggestions. Thanks!!

Comment (dismirlian) [reply src/core/thread/network_data_local.cpp]: I see why it failed the tests (the ...OrExit macros are hard to get used to...). Implemented a minor variation your suggestion, so as not to mix bitwise and logical ORs and allow for lazy evaluation. Thanks for your help!

**Pull request merged**